helmfile/charts/s3-exporter/templates/servicemonitors.yaml

```diff
@@ -0,0 +1,42 @@
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "s3-exporter.fullname" . }}
+  labels:
+    {{- include "s3-exporter.labels" . | nindent 4 }}
+spec:
+  endpoints:
+  # One endpoint per bucket so that we can pass the name of the bucket as param
```

Comment:
Why is this necessary?

Reply:
I saw that this is a requirement in the repo. So I guess this is the way it is.

Reply:
The s3_exporter works by "probes", so we need to tell Prometheus what to probe for. With curl, you do it like this:
```
curl localhost:9340/probe?bucket=some-bucket&prefix=some-folder/some-file.txt
```
We want to probe all our buckets so we need to somehow tell Prometheus to scrape the s3-exporter for all of them. Specifying multiple endpoints was the best solution I could come up with. The blackbox exporter [is very similar](https://github.com/elastisys/compliantkubernetes-apps/blob/main/helmfile/charts/blackbox/templates/servicemonitors.yaml), but there we do one whole ServiceMonitor per target instead of just an endpoint. Could probaby rework it.